lib/chef/provisioning/convergence_strategy/install_sh.rb

```diff
@@ -29,58 +30,19 @@ def initialize(convergence_options, config)
 
-      def install_sh_command_line
-        arguments = install_sh_arguments ? " #{install_sh_arguments}" : ""
-        arguments << " -v #{chef_version}" if chef_version
-        arguments << " -p" if prerelease
-        "bash -c '#{bootstrap_env} bash #{install_sh_path}#{arguments}'"
-      end
-
       def setup_convergence(action_handler, machine)
         super
 
-        # Check for existing chef client.
-        version = machine.execute_always('chef-client -v')
-
-        # Don't do install/upgrade if a chef client exists and
-        # no chef version is defined by user configs or
-        # the chef client's version already matches user config
-        if version.exitstatus == 0
-          version = version.stdout.strip
-          if !chef_version
-            return
-          # This logic doesn't cover the case for a client with 12.0.1.dev.0 => 12.0.1
-          # so we decided to just use exact version for the time being (see comments in PR 303)
-          #elsif version.stdout.strip =~ /Chef: #{chef_version}([^0-9]|$)/
-          elsif version =~ /Chef: #{chef_version}$/
-            Chef::Log.debug "Already installed chef version #{version}"
-            return
-          elsif version.include?(chef_version)
-            Chef::Log.warn "Installed chef version #{version} contains desired version #{chef_version}.  " +
-              "If you see this message on consecutive chef runs tighten your desired version constraint to prevent " +
-              "multiple convergence."
-          end
+        opts = {"prerelease" => prerelease}
+        if convergence_options[:bootstrap_proxy]
+          opts["http_proxy"] = convergence_options[:bootstrap_proxy]
+          opts["https_proxy"] = convergence_options[:bootstrap_proxy]
         end
 
-        # Install chef client
-        # TODO ssh verification of install.sh before running arbtrary code would be nice?
-        if !convergence_options[:bootstrap_proxy] || convergence_options[:bootstrap_proxy].empty?
-          @@install_sh_cache[install_sh_url] ||= Net::HTTP.get(URI(install_sh_url))
-        else
-          @@install_sh_cache[install_sh_url] ||= begin
-            proxy_uri = URI.parse(convergence_options[:bootstrap_proxy])
-            chef_uri  = URI.parse(@install_sh_url)
-            proxy     = Net::HTTP::Proxy(proxy_uri.host, proxy_uri.port, proxy_uri.user, proxy_uri.password)
-            req       = Net::HTTP::Get.new(chef_uri.path)
-            script    = proxy.start(chef_uri.host, :use_ssl => proxy_uri.scheme == 'https') do |http|
-              http.request(req)
-            end
-            script.body
-          end
-        end
-        machine.write_file(action_handler, install_sh_path, @@install_sh_cache[install_sh_url], :ensure_dir => true)
-        # TODO handle bad version case better
-        machine.execute(action_handler, install_sh_command_line)
+        install_command = Mixlib::Install.new(chef_version, false, opts).install
```

Comment:
`install` seems like a misleading method name if all it does is return a string that then needs to be executed to perform the actual install.